Unity-Technologies/Graphics
# Shadergraph/bugfix 1186867

### Checklist for PR maker
- [x ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.

---
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?
There was a bug in the Shader Graph import process which would treat warnings and errors as both being shader-breaking errors and thus a warning in the graph would cause the fallback error shader to display.
---
### Testing status

**Manual Tests**: What did you do?
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
Added automated tests to the test project to verify warnings will not break shaders while errors still will


**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline
